### PR TITLE
[storage] [backend] Add Handle to BE object now

### PR DESCRIPTION
**Pull request by glimchb**
Related to #174

Resource-oriented design https://google.aip.dev/121

Signed-off-by: Boris Glimcher <Boris.Glimcher@emc.com>

### CODE REVIEW

File: storage/v1alpha1/backend_nvme_tcp.proto [hunk 1]
```diff
@@ -39,49 +44,37 @@ message NVMfRemoteControllerConnectRequest {
     NVMfRemoteController ctrl = 1;
```
Comment (mkalderon) [inline storage/v1alpha1/backend_nvme_tcp.proto:44]: why not provide the objectKey here ?

Comment (glimchb) [reply storage/v1alpha1/backend_nvme_tcp.proto:44]: good question. according to what we did in frontend, we put the ID inside the object, so I wanted to align those here as well. From https://google.aip.dev/133#user-specified-ids indeed IDs should be in the request message, not the object and named `string {resource}_id`

File: storage/v1alpha1/backend_nvme_tcp.proto [hunk 2]
```diff
@@ -39,49 +44,37 @@ message NVMfRemoteControllerConnectRequest {
...
-    // Intentionally empty.
+    // object's unique identifier
+    common.v1.ObjectKey id = 1;
 }
 
 message NVMfRemoteControllerListRequest {
-    int64 id = 1;
+    int32 page_size = 1;
+    string page_token = 2;
 }
```
[the remark this answers is not in the document]
Comment (glimchb) [reply storage/v1alpha1/backend_nvme_tcp.proto:60]: I could've split in 2 commits here, you are right.
the page fields are from https://google.aip.dev/132
just aligning the objects here from the frontend work we did

**Comment by jainvipin** [comment]
we could go with using keys/handles and status/spec on backend objects as well.

File: storage/v1alpha1/backend_aio.proto [hunk 3]
```diff
@@ -11,17 +11,40 @@ option java_outer_classname = "BackendAioProto";
...
 service AioControllerService {
-    rpc AioControllerCreate (AioControllerCreateRequest) returns (AioController) {}
-    rpc AioControllerDelete (AioControllerDeleteRequest) returns (google.protobuf.Empty) {}
-    rpc AioControllerGet    (AioControllerGetRequest)    returns (AioController)    {}
-    rpc AioControllerGetList   (AioControllerGetListRequest)   returns (AioControllerList)   {}
-    rpc AioControllerGetStats  (AioControllerGetStatsRequest)  returns (AioControllerStats)  {}
-    rpc AioControllerUpdate (AioControllerUpdateRequest) returns (AioController) {}
+    rpc CreateAioController (CreateAioControllerRequest) returns (AioController) {
+        option (google.api.http) = {
+            post: "/v1/subsystems"
```
Comment (glimchb) [inline storage/v1alpha1/backend_aio.proto:22]: URLs will be fixed #188

File: storage/v1alpha1/backend_nvme_tcp.proto [hunk 4]
```diff
@@ -10,18 +10,47 @@ option java_outer_classname = "BackendNvmeTcpProto";
...
 // Back End (network-facing) APIs. NVMe/TCP and NVMe/RoCEv2 protocols are covered by this service.
 service NVMfRemoteControllerService {
-    rpc NVMfRemoteControllerConnect (NVMfRemoteControllerConnectRequest) returns (NVMfRemoteControllerConnectResponse) {}
-    rpc NVMfRemoteControllerDisconnect (NVMfRemoteControllerDisconnectRequest) returns (NVMfRemoteControllerDisconnectResponse) {}
-    rpc NVMfRemoteControllerReset (NVMfRemoteControllerResetRequest) returns (NVMfRemoteControllerResetResponse) {}
-    rpc NVMfRemoteControllerList   (NVMfRemoteControllerListRequest)   returns (NVMfRemoteControllerListResponse)   {}
-    rpc NVMfRemoteControllerGet    (NVMfRemoteControllerGetRequest)    returns (NVMfRemoteControllerGetResponse)    {}
+    rpc CreateNVMfRemoteController (CreateNVMfRemoteControllerRequest) returns (NVMfRemoteController) {
+        option (google.api.http) = {
+            post: "/v1/subsystems"
```
Comment (glimchb) [inline storage/v1alpha1/backend_nvme_tcp.proto:21]: URLs will be fixed #188

**Review by sandersms — APPROVED** [review]
Changes look good to me

**Pull request merged**